one-ui/src/main/java/com/marklogic/hub/curation/controllers/AbstractArtifactController.java

```diff
@@ -46,4 +64,62 @@ protected ArtifactService getArtifactService() {
     }
 
     protected abstract String getArtifactType();
+
+    protected void writeArtifact(ObjectNode artifact) throws IOException {
+        Path fileLocation = buildArtifactProjectLocation(getNameFromArtifact(artifact), getVersionFromArtifact(artifact));
+        // create folders if needed
+        if (!fileLocation.getParent().toFile().exists()) {
+            fileLocation.getParent().toFile().mkdirs();
+        }
+        Files.write(fileLocation, artifact.toPrettyString().getBytes());
+    }
+
+    protected void deleteArtifact(String artifactName, String artifactVersion) throws IOException {
+        Path fileLocation = buildArtifactProjectLocation(artifactName, artifactVersion);
+        Files.deleteIfExists(fileLocation);
+    }
+
+    protected String getNameFromArtifact(ObjectNode artifact) {
+        ObjectNode artifactTypeInfo = getArtifactTypeInfo();
+        return artifact.get(artifactTypeInfo.get("nameProperty").asText()).asText();
+    }
+
+    protected String getVersionFromArtifact(ObjectNode artifact) {
```

Comment:
Do we have any use case for versions yet? Or will the value just be "1" for everything until we have a use case for separate versions of an artifact?

Reply:
We don't yet. It is more in preparation for when we move mapping to this new pattern. Right now the version is always null and the artifact type info won't provide a version property, so it isn't taken into account.